pkg/render/manager_route_config.go

```diff
@@ -0,0 +1,396 @@
+
+	routesAnnotationKey = "hash.operator.tigera.io/route-configuration"
+)
+
+type tlsTerminatedRoute struct {
+	// Destination is the destination URL
+	Destination string `json:"destination"`
+	// Path is the path portion of the URL based on which we proxy
+	Path string `json:"path"`
+
+	// CABundlePath is where we read the CA bundle from to authenticate the
+	// destination (if non-empty)
+	CABundlePath string `json:"caBundlePath,omitempty"`
+	// PathRegexp, if not nil, checks if Regexp matches the path
+	PathRegexp string `json:"pathRegexp,omitempty"`
+	// PathReplace if not nil will be used to replace PathRegexp matches
+	PathReplace string `json:"pathReplace,omitempty"`
+	// AllowInsecureTLS allows https with insecure tls settings
+	AllowInsecureTLS bool `json:"allowInsecureTLS,omitempty"`
+
+	// ClientCertPath and ClientKeyPath can be set for mTLS on the connection
+	// from Voltron to the destination.
+	ClientCertPath string `json:"clientCertPath,omitempty"`
+	ClientKeyPath  string `json:"clientKeyPath,omitempty"`
+
+	Unauthenticated bool `json:"unauthenticated,omitempty"`
+}
+
+type tlsPassThroughRoute struct {
+	// Destination is the destination URL
+	Destination string `json:"destination"`
+
+	// ServerName
+	ServerName string `json:"serverName"`
+}
+
+// VoltronRouteConfigBuilder is an interface that provides methods to build a VoltronRouteConfig.
+// Implementations of this interface should provide methods to add different types of routes,
+// such as TLSTerminatedRoute and TLSPassThroughRoute, as well as methods to add ConfigMaps
+// and Secrets to the route configuration to generate annotations that change when the contents of
+// objects change (used to restart Voltron). The Build method should be used to create the final
+// VoltronRouteConfig object.
+type VoltronRouteConfigBuilder interface {
+	AddTLSTerminatedRoute(routes operatorv1.TLSTerminatedRoute)
+	AddTLSPassThroughRoute(routes operatorv1.TLSPassThroughRoute)
+	AddConfigMap(configMap *corev1.ConfigMap)
+	AddSecret(secret *corev1.Secret)
+	Build() (*VoltronRouteConfig, error)
+}
+
+type voltronRouteConfigBuilder struct {
+	mountedConfigMaps map[string]struct{}
+	mountedSecrets    map[string]struct{}
+
+	volumeMounts    []corev1.VolumeMount
+	volumes         []corev1.Volume
+	hashableContent []interface{}
+
+	configMaps map[string]*corev1.ConfigMap
+	secrets    map[string]*corev1.Secret
+
+	tlsTerminatedRoutes  []operatorv1.TLSTerminatedRoute
+	tlsPassThroughRoutes []operatorv1.TLSPassThroughRoute
+}
+
+func NewVoltronRouteConfigBuilder() VoltronRouteConfigBuilder {
+	return &voltronRouteConfigBuilder{
+		configMaps:        map[string]*corev1.ConfigMap{},
+		secrets:           map[string]*corev1.Secret{},
+		mountedConfigMaps: map[string]struct{}{},
+		mountedSecrets:    map[string]struct{}{},
+	}
+}
+
+func (builder *voltronRouteConfigBuilder) AddConfigMap(configMap *corev1.ConfigMap) {
+	if _, ok := builder.configMaps[configMap.Name]; !ok {
```

Comment:
I was thinking that we just require that whatever is configuring the routes can put the config maps in the namespace for voltron, otherwise this operator just needs to copy more resources in the cluster, and I'm not sure what the benefit is. I'm not sure if there's a security concern, it might be that the other operators load tls routes / secrets that shouldn't be loaded into the operator, but that's likely already the case since the other operators need access to the tigera-operator namespace.